sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLListenerSuite.scala

```diff
@@ -343,6 +343,8 @@ class SQLListenerMemoryLeakSuite extends SparkFunSuite {
       .set("spark.sql.ui.retainedExecutions", "50") // Set it to 50 to run this test quickly
     val sc = new SparkContext(conf)
     try {
+      // Clear the sql listener created by a previous test suite.
+      SQLContext.clearSqlListener()
```

Comment:
This is not a public API. So the user cannot clear `SQLContext.sqlListener`? This will be a memory leak considering `SQLListener` usually stores a lot of data.